task2/script.js

```diff
@@ -0,0 +1,156 @@
+        director: "David Yates",
+        writer: "Steve Kloves, J.K. Rowling",
+        actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+        plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort\'s dark past.",
+        country: "United Kingdom",
+        poster: "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 492245,
+        boxOffice: "$302,305,431",
+        production: "Heyday Films, Warner Bros.",
+    },
+    {
+        id: 5,
+        title: "Harry Potter and the Sorcerer's Stone",
+        year: 2001,
+        released: "16 Nov 2001",
+        runtime: "152 min",
+        genre: ["Adventure", "Family", "Fantasy"],
+        director: "Chris Columbus",
+        writer: "J.K. Rowling, Steve Kloves",
+        actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+        plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+        country: "United Kingdom, United States",
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+]
+
+//Задача 1 
+//1. Собрать в массив все жанры фильмов (без повторения)
+const genre = arr.map(function (a) { return a.genre; }).join(',').split(',');
+const genreArr = Array.from(new Set(genre));
+console.log(genreArr)
+
+//Задача 2
+//Собрать в массив всех актеров всех фильмов (без повторения)
+const actor = arr.map(function (a) { return a.actors }).join(',').split(',');
+const actorArr = Array.from(new Set(actor));
+console.log(actorArr)
+
+//Задача 3 
+//Отсортировать фильмы по рейтингу по убыванию
+const sortFilm = arr.sort((a, b) => a.imdbRating < b.imdbRating ? 1 : -1)
+console.log(sortFilm);
+
+//Задача 4
+//Создать новый массив, где объекты фильмов будут состоять из следующих полей: id, title, released, plot
+const array = arr.map(({ id, title, released, plot }) => ({ id, title, released, plot }))
+console.log(array);
+
+//Задача 6
+//Создать массив авторов (поле writer) без повторений.
+const writer = arr.map(function (a) { return a.writer }).join(',').replace(/\s*,\s*/g, ",").split(',')
+const writerArr = Array.from(new Set(writer));
+console.log(writerArr)
+
+//Задача 7
+//Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть новый отфильтрованный массив, с фильмами, где строка входит в название фильма.
+function getFilms(arr, str) {
+    const film = arr.filter(item => item.title.split(" ").includes(str))
```

Comment:
you can do it without variable

getFilms(arr, str) = arr.filter(item => item.title.split(" ").includes(str))